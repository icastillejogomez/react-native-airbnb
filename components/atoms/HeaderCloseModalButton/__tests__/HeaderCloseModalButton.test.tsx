import { render } from '@testing-library/react-native'

import { HeaderCloseModalButton } from '..'

describe('HeaderCloseModalButton', () => {
  test('renders without crashing', () => {
    const tree = render(<HeaderCloseModalButton />)
    expect(tree).toMatchSnapshot()
  })

  test('run onPress handler when pressed', () => {
    const onPress = jest.fn()
    const tree = render(<HeaderCloseModalButton onPress={onPress} />)
    tree.UNSAFE_getByType(HeaderCloseModalButton).props.onPress()
    expect(onPress).toHaveBeenCalledTimes(1)
  })
})
