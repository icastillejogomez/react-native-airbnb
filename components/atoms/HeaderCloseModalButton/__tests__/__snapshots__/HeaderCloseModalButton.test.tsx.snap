// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderCloseModalButton renders without crashing 1`] = `null`;
